Disable hardware topology tests. The failure of these tests is bening.
See https://mailman-1.sys.kth.se/pipermail/gromacs.org_gmx-users/2020-March/128857.html
author: Alex Domingo (Vrije Universiteit Brussel)
--- src/gromacs/hardware/tests/CMakeLists.txt.orig	2021-01-04 15:03:40.808453000 +0100
+++ src/gromacs/hardware/tests/CMakeLists.txt	2021-01-04 15:05:46.544205000 +0100
@@ -33,5 +33,4 @@
 # the research papers on the package. Check out http://www.gromacs.org.
 
 gmx_add_unit_test(HardwareUnitTests hardware-test
-                  cpuinfo.cpp
-                  hardwaretopology.cpp)
+                  cpuinfo.cpp)
